activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -733,7 +733,7 @@ def register_integer_type(mapping, key, options) # :nodoc:
       end
 
       def extract_precision(sql_type)
-        if /datetime/ === sql_type
+        if /^(?:date)?time/ === sql_type
```

Comment:
It seems the missing ^ anchor not intentional.

If you prefer to shortest pattern, it is possible to match `/time/`. It is only datetime, timestamp and time to match this in MySQL. These types can have a precision.

http://dev.mysql.com/doc/refman/5.6/en/fractional-seconds.html